app/notification_email.go

```diff
@@ -283,6 +287,36 @@ func getFormattedPostTime(user *model.User, post *model.Post, useMilitaryTime bo
 	}
 }
 
+func (a *App) generateHyperlinkForChannels(postMessage, teamName, teamURL string) string {
+	var publicChannels *model.ChannelList
+	if team, err := a.GetTeamByName(teamName); err != nil {
+		mlog.Error(fmt.Sprintf("Encountered error while looking up team by name for %s", teamName))
+		return postMessage
+	} else {
+		if publicChannels, err = a.GetPublicChannelsForTeam(team.Id, 0, 100); err != nil {
+			mlog.Error(fmt.Sprintf("Encountered error while looking up Public Channels for Team: %s", teamName))
+			return postMessage
+		}
+	}
+
+	r := regexp.MustCompile(`\~[\w-]+\b`)
+	matches := r.FindAllString(postMessage, -1)
+	var channelURL, channelHyperLink string
+	channelNameMap := make(map[string]struct{})
+	for _, channelName := range matches {
+		channelNameMap[channelName[1:]] = struct{}{}
+	}
+
+	for _, ch := range *publicChannels {
+		if _, ok := channelNameMap[ch.DisplayName]; ok {
```

Comment:
This should be using the channel's `Name` when looking for links, not it's `DisplayName`